### PR TITLE
Created samples for StochasticGradientDescentNonCalibrated learner.

### PR DESCRIPTION
**Pull request by zeahmed**
Created samples for StochasticGradientDescentNonCalibrated learner exposed via following methods.

https://github.com/dotnet/machinelearning/blob/a0edc5c5724ed2895fd0c2fa5c2525d03de13455/src/Microsoft.ML.StandardLearners/StandardLearnersCatalog.cs#L82

https://github.com/dotnet/machinelearning/blob/a0edc5c5724ed2895fd0c2fa5c2525d03de13455/src/Microsoft.ML.StandardLearners/StandardLearnersCatalog.cs#L102

### CODE REVIEW

File: docs/samples/Microsoft.ML.Samples/Dynamic/Trainers/BinaryClassification/StochasticGradientDescentNonCalibrated.cs [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+        // For more details about this dataset, please see https://archive.ics.uci.edu/ml/datasets/adult.
+        public static void Example()
+        {
+            // Create a new context for ML.NET operations. It can be used for exception tracking and logging, 
+            // as a catalog of available operations and as the source of randomness.
+            // Setting the seed to a fixed number in this example to make outputs deterministic.
+            var mlContext = new MLContext(seed: 0);
+
+            // Download and featurize the dataset.
+            var data = SamplesUtils.DatasetUtils.LoadFeaturizedAdultDataset(mlContext);
```
Comment (wschin) [inline docs/samples/Microsoft.ML.Samples/Dynamic/Trainers/BinaryClassification/StochasticGradientDescentNonCalibrated.cs]: It'd nice if you can use in-memory data for trainers. Such a way has some benefits as discussed in #2726.

File: docs/samples/Microsoft.ML.Samples/Dynamic/Trainers/BinaryClassification/StochasticGradientDescentNonCalibrated.cs [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+
+            // Create data training pipeline.
+            var pipeline = mlContext.BinaryClassification.Trainers.StochasticGradientDescentNonCalibrated();
+
+            // Fit this pipeline to the training data.
+            var model = pipeline.Fit(trainTestData.TrainSet);
+
+            // Evaluate how the model is doing on the test data.
+            var dataWithPredictions = model.Transform(trainTestData.TestSet);
+            var metrics = mlContext.BinaryClassification.EvaluateNonCalibrated(dataWithPredictions);
```
Comment (wschin) [inline docs/samples/Microsoft.ML.Samples/Dynamic/Trainers/BinaryClassification/StochasticGradientDescentNonCalibrated.cs]: Need to show how user can inspect the prediction. Evaluation is not the major part in products.

File: docs/samples/Microsoft.ML.Samples/Dynamic/Trainers/BinaryClassification/StochasticGradientDescentNonCalibrated.cs (hunk 1 above)
Comment (shmoradims) [reply docs/samples/Microsoft.ML.Samples/Dynamic/Trainers/BinaryClassification/StochasticGradientDescentNonCalibrated.cs]: Hi Wei-Sheng, please mark your in-memory comments as optional, until #2726 reaches a conclusion. #2726 is still in discussion and there's no clear actionable change that everybody agrees on, yet. 

---
In reply to: [261020122](https://github.com/dotnet/machinelearning/pull/2770#discussion_r261020122) [](ancestors = 261020122)

Comment (zeahmed) [reply docs/samples/Microsoft.ML.Samples/Dynamic/Trainers/BinaryClassification/StochasticGradientDescentNonCalibrated.cs]: Lets do these things once we have decision. 

---
In reply to: [261770052](https://github.com/dotnet/machinelearning/pull/2770#discussion_r261770052) [](ancestors = 261770052,261020122)

File: docs/samples/Microsoft.ML.Samples/Dynamic/Trainers/BinaryClassification/StochasticGradientDescentNonCalibrated.cs (hunk 2 above)
Comment (zeahmed) [reply docs/samples/Microsoft.ML.Samples/Dynamic/Trainers/BinaryClassification/StochasticGradientDescentNonCalibrated.cs]: Lets do these things once we have decision. 

---
In reply to: [261020336](https://github.com/dotnet/machinelearning/pull/2770#discussion_r261020336) [](ancestors = 261020336)

File: src/Microsoft.ML.StandardLearners/StandardLearnersCatalog.cs [hunk 3]
```diff
@@ -79,6 +79,13 @@ public static class StandardLearnersCatalog
         /// <param name="maxIterations">The maximum number of passes through the training dataset; set to 1 to simulate online learning.</param>
         /// <param name="initLearningRate">The initial <a href="tmpurl_lr">learning rate</a> used by SGD.</param>
         /// <param name="l2Weight">The L2 weight for <a href='tmpurl_regularization'>regularization</a>.</param>
+        /// <example>
+        /// <format type="text/markdown">
+        /// <![CDATA[
+        /// [!code-csharp[StochasticGradientDescentWithOptions](~/../docs/samples/docs/samples/Microsoft.ML.Samples/Dynamic/Trainers/BinaryClassification/StochasticGradientDescentNonCalibrated.cs)]
```
Comment (shmoradims) [inline src/Microsoft.ML.StandardLearners/StandardLearnersCatalog.cs]: >StochasticGradientDescentWithOptions [](start = 26, length = 36)

please fix this copy-paste error.

File: src/Microsoft.ML.StandardLearners/StandardLearnersCatalog.cs [hunk 4]
```diff
@@ -99,6 +106,13 @@ public static class StandardLearnersCatalog
         /// </summary>
         /// <param name="catalog">The binary classification catalog trainer object.</param>
         /// <param name="options">Trainer options.</param>
+        /// /// <example>
+        /// <format type="text/markdown">
+        /// <![CDATA[
+        /// [!code-csharp[StochasticGradientDescentWithOptions](~/../docs/samples/docs/samples/Microsoft.ML.Samples/Dynamic/Trainers/BinaryClassification/StochasticGradientDescentNonCalibratedWithOptions.cs)]
```
Comment (shmoradims) [inline src/Microsoft.ML.StandardLearners/StandardLearnersCatalog.cs]: >StochasticGradientDescentWithOptions [](start = 26, length = 36)

same here

**Pull request merged**